publisher/pubsub.go

```diff
@@ -0,0 +1,234 @@
+package publisher
```

Comment:
@turtleDev let's create a dedicated package for each publisher within publisher folder.

Reply:
done.